app/scripts/migrations/089.ts

```diff
@@ -0,0 +1,40 @@
+import { cloneDeep } from 'lodash';
+import { hasProperty, isObject } from '@metamask/utils';
+
+export const version = 89;
+
+/**
+ * Explain the purpose of the migration here.
+ *
+ * @param originalVersionedData - Versioned MetaMask extension state, exactly what we persist to dist.
+ * @param originalVersionedData.meta - State metadata.
+ * @param originalVersionedData.meta.version - The current state version.
+ * @param originalVersionedData.data - The persisted MetaMask state, keyed by controller.
+ * @returns Updated versioned MetaMask extension state.
+ */
+export async function migrate(originalVersionedData: {
+  meta: { version: number };
+  data: Record<string, unknown>;
+}) {
+  const versionedData = cloneDeep(originalVersionedData);
+  versionedData.meta.version = version;
+  versionedData.data = transformState(versionedData.data);
+  return versionedData;
+}
+
+function transformState(state: Record<string, unknown>) {
+  if (
+    !hasProperty(state, 'PhishingController') ||
+    !isObject(state.PhishingController) ||
+    !hasProperty(state.PhishingController, 'listState')
+  ) {
+    return state;
+  }
+
+  const { PhishingController } = state;
+
+  PhishingController.phishingLists = [PhishingController.listState];
```

Comment:
There was a change made to the phishing list type as part of this release as well: a `name` property was added (see https://github.com/MetaMask/core/pull/1123/files#diff-73fbecd61d9f85b57fee9c8377652f5563d90e026e54fa105ca91fcc2cbc076fR66 ). This is used for attribution.


With the pre-existing phishing config, we have no way to know which name to apply. We could set it to MetaMask, but it includes entries from both MetaMask and Phishfort.

Perhaps we should delete pre-existing phishing list config instead? It will be correctly refreshed on the next update.

Reply:
We only really need to delete the `listState` configuration. Perhaps deleting the timestamps to force an immediate update would be a good idea too. But we'll want to preserve the `whitelist`, that is a user-curated list of trusted origins that is still in the correct format with this new update.